addons/product/models/product_template.py

```diff
@@ -628,6 +628,10 @@ def _create_variant_ids(self):
 
         if variants_to_activate:
             variants_to_activate.write({'active': True})
+        if exclusion:
+            for variant in variants_to_activate:
+                if not self._is_combination_possible(variant.product_template_attribute_value_ids):
```

Comment:
Note: `_is_combination_possible`  is ensure one, it cannot be called on self without iterating each template.

But as I'm suggesting to move the code above, it will fall in the existing loop automatically.